R/dominantTaxa.R

```diff
@@ -55,31 +41,45 @@
 #' x <- GlobalPatterns
 #'
 #' # Finds the dominant taxa.
-#' x <- dominantTaxa(x)
+#' x <- addDominantTaxa(x)
 #' # Information is stored to colData
 #' colData(x)
-#' # Gets the overview of dominant taxa
-#' overview <- getDominantTaxa(x)
-#' overview
 #'
 #' # If taxonomic information is available, it is possible to find the most
 #' # dominant group from specific taxonomic level, here family level. The name
 #' # of column can be specified.
-#' x <- dominantTaxa(x, rank="Family", name="dominant_taxa_ranked_with_family")
+#' x <- addDominantTaxa(x, rank="Family", name="dominant_taxa_ranked_with_family")
 #' colData(x)
-#' # Gets the overview of dominant taxa
-#' overview <- getDominantTaxa(x)
-#' overview
 #'
 #' x <- microbiomeDataSets::dietswap()
-#' x <- dominantTaxa(x)
+#' x <- addDominantTaxa(x)
 #' colData(x)
-#' # With group, it is possible to group observations based on groups specified
-#' # Gets the overview of dominant taxa
-#' overview <- getDominantTaxa(x, group = "nationality")
-#' overview
+#' 
 NULL
 
+#' @rdname dominantTaxa
+#' @export
+setGeneric("addDominantTaxa",signature = c("x"),
+           function(x,
+                    abund_values = "counts",
+                    rank = NULL,
+                    name = "dominant_taxa")
+               standardGeneric("addDominantTaxa"))
+
+#' @rdname dominantTaxa
+#' @export
+setMethod("addDominantTaxa", signature = c(x = "SummarizedExperiment"),
+          function(x,
+                abund_values = "counts",
+                rank = NULL,
+                name = "dominant_taxa"){
+              dominantTaxa(x, 
+                           abund_values = abund_values,
+                           rank = rank,
+                           name = name)
```

Comment:
I am not sure, but this wasn't the idea, was it?

From the discussion I think the following requirements need to be met:
-  `dominantTaxa` returns a `DataFrame`
- `addDominantTaxa` returns the input `x`, with output of `dominantTaxa` added to `colData`

This allows the used to decided, whether he/she/they need the data separate or combined within a `SE`

PS: test also need to be changed accordingly.

Reply:
You make good points here and we need to rethink this. My idea is that,  

- `dominantTaxa` should be `perSampleDominantTaxa` and return per sample the dominant taxa, like below
![image](https://user-images.githubusercontent.com/16476233/120201215-7b1cde00-c225-11eb-857b-9ac9031db1be.png)

- `addPerSampleDominantTaxa` should add the output of `perSampleDominantTaxa`  to `colData`
- While `summarizeDominantTaxa` / ` countDominantTaxa` returns a summary of dominant taxa based on its dominance in all samples 
![image](https://user-images.githubusercontent.com/16476233/120201878-32195980-c226-11eb-89f5-3275540d9680.png)

or by a group of samples.

![image](https://user-images.githubusercontent.com/16476233/120201311-9851ac80-c225-11eb-9d63-b655bb47ac65.png)

I think this is a more logical approach. Do you agree?

PS: looking at the outputs above we also need to think of how best to deal with NAs especially when working with taxonomic ranks from family to species level...

Reply:
This is a good summary. So `dominantTaxa` should not return a `DataFrame`, but a named `character` vector. (Currently `dominantTaxa` and `addDominantTaxa` return the same result, aka. as a modified SE. So this needs to be fixed.)

I think you can add an `na.rm` argument. This is expected by the user, because thats exactly how the argument is named for other functions. `agglomerateBy*` uses it as well.